CONTRIBUTING.md

```diff
@@ -1,106 +1,186 @@
- - `<RPName>\<packageName>\tests\Microsoft.Azure.Management.<RPName>.Tests.csproj`
+3. Project names helps in using basic heuristics in finding projects as well it's associated test projects during CI process.
+4. Create a new directory using the name of your service/ resource provider as specified in [azure-rest-api-specs/specification](https://github.com/Azure/azure-rest-api-specs/tree/master/specification) Repo
+5. Follow the the directory structure below
+
+```
+sdk\<service name>\<package name>\README.md
+sdk\<service name>\<package name>\*src*
+sdk\<resource provider name>\<package name>\*tests*
+sdk\<resource provider name>\<package name>\*samples*
+```
+
+e.g.
+
+```
+sdk\eventgrid\Microsoft.Azure.EventGrid\src\Microsoft.Azure.EventGrid.csproj
+sdk\eventgrid\Microsoft.Azure.EventGrid\tests\Microsoft.Azure.EventGrid.Tests.csproj
+sdk\eventgrid\Microsoft.Azure.Management.EventGrid\src\Microsoft.Azure.Management.EventGrid.csproj
+sdk\eventgrid\Microsoft.Azure.Management.EventGrid\tests\Microsoft.Azure.Management.EventGrid.Tests.csproj
+```
+
+> \*Ensure that your service name is the same as it is specified in the [azure-rest-api-specs/specification](https://github.com/Azure/azure-rest-api-specs/tree/master/specification) Repo, that your csproj files starts with **Microsoft.Azure\***
+> , that test files end with **.Tests** and that management plane project files containt **.Management.**
+
 7. Copy .csproj from any other .csproj and update the following information in the new .csproj
- - PackageId
- - Description
- - AssemblyTitle
- - AssemblyName
- - Version
- - PackageTags
- - PackageReleaseNotes (this is important because this information is displayed on www.nuget.org when your nuget package is published
-8. Copy existing generate.ps1 file from another dirctory and update the `ResourceProvider` name that is applicable to your SDK. Resource provider refers to the relative path of your REST spec directory in Azure-Rest-Api-Specs repository
-During SDK generation, this path helps to locate the REST API spec from the `https://github.com/Azure/azure-rest-api-specs`
+
+   | For DataPlane       | For Management Plane |
+   | ------------------- | -------------------- |
+   | AssemblyTitle       | PackageId            |
+   | Description         | Description          |
+   | VersionPrefix       | AssemblyTitle        |
+   | PackageTags         | AssemblyName         |
+   | PackageReleaseNotes | Version              |
+   |                     | PackageTags          |
+   |                     | PackageReleaseNotes  |
+
+> PackageReleaseNotes are important because this information is displayed on www.nuget.org when your nuget package is published
+
+8. Copy existing generate.ps1 file from another service and update the `ResourceProvider` name that is applicable to your SDK. Resource provider refers to the relative path of your REST spec directory in Azure-Rest-Api-Specs repository
```

Comment:
@shahabhijeet we still need to fix the psmodules if we want people to use generate.ps1.